### DISCUSSION TITLE
Running top-level LFRic tests on Monsoon by a non-MO user

### DISCUSSION THREAD

**Discussion by dennissergeev** [MetOffice/simulation-systems#148] [Monsoon]
Hello,

I wanted to run standard tests (`make test-suite`) on my branch on Monsoon - could you help me to set this up correctly please? I don't have access to SPICE or meto-xc40.

Following the [LFRicTechnical wiki](https://code.metoffice.gov.uk/trac/lfric/wiki/LFRicTechnical/Testing/SystemTesting#TestSuiteonMONSooN), I tried connecting to `exvmsrose` but this didn't work. I then tried loading the `meto-common-environment` on `xcslc0` anyway - this worked for `meto-common-environment/lfric/17` (not `dynamo`).  This allowed me to launch 
```bash
make test-suite TEST_SUITE_TARGETS=monsoon-xc40
```
and some tests have even succeeded. However, many failed to launch due to a missing optional configuration, `monsoon-xc40`. I suppose this is expected but I would still appreciate any advice on how to run these tests properly.

Note that `rose-stem` tests, e.g. `gungho` and `lfric_atm` run OK.

Thanks!

**Comment by stevemullerworth** [comment]
Hi Denis

Sorry, the dynamo reference is out of date - we will fix this. The correct place to look is the DevelopmentEnvironment page:

https://code.metoffice.gov.uk/trac/lfric/wiki/DevelopmentEnvironment

Configurations were provided where they were required, so I think only lfric_atm and gungho_model would work unless the configuration settings for Monsoon are added. `grep -ir monsoon` on the gungho_model directory might tell you what needs copying to other apps, but I've not tested it (I don't have Monsoon access). There would also need to be KGOs, but they can probably be symlinked to the XCS ones as is done for gungho_model and lfric_atm.

**Reply by dennissergeev** [reply]
Thanks @stevemullerworth! Could you also remind me how to run the code style checker please? (which I hope is possible on Monsoon?..)

**Comment by james-bruten-mo** [comment]
It's probably worth noting that in a few weeks lfric will be split into apps and core with a new test suite for lfric_apps. As currently, only lfric_atm and gungho_model work on Monsoon. You can launch it from Monsoon by running:
```
export CYLC_VERSION=8
rose stem --group=monsoon
cylc play <name-of-suite>
```
which will only run lfric_atm and gungho_model.
Again, it's not there yet - will be ready once the lfric repo split has occured.

**Comment by stevemullerworth** [comment]
Unfortunately, for pragmatic and technical reasons (Python versions), `stylist` is not available on Monsoon. You can install it on a suitable platform 

https://github.com/MetOffice/stylist/

To run it on um_physics, say:
$ cd um_physics
$ stylist -configuration ../etc/stylist.py source

You can add the -v option for verbose mode which lists the files and the checks it is applying.

**Reply by dennissergeev** [reply]
Many thanks @stevemullerworth!
